### PR TITLE
Opi api changes

### PR DESCRIPTION
**Pull request by jainvipin**
adopted the spdk client and server frontned apis to use new opi-frontend apis
the bridge code keeps cache of various objects, this allows for key association and return information on other situations. I have kept the code to query from spdk server from the bridge, to be sure that backend system has all the relevant information as well
pls review @glimchb

### CODE REVIEW

File: client/frontend.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"log"
 
+	pbc "github.com/opiproject/opi-api/common/v1/gen/go"
 	pb "github.com/opiproject/opi-api/storage/v1/gen/go"
 	"google.golang.org/grpc"
 )
```
Comment (glimchb) [inline client/frontend.go:10]: looks good. optionally we can use `uuid.New().String()` for the `ObjectKey` just to keep things random and interesting... 
not must for this PR

File: server/frontend.go [hunk 2]
```diff
@@ -106,46 +123,70 @@ func (s *server) NVMeSubsystemStats(ctx context.Context, in *pb.NVMeSubsystemSta
...
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+
 	params := NvmfSubsystemAddNsParams{
-		Nqn: in.GetNamespace().GetSubsystemId(),
+		Nqn: subsys.Nqn,
 	}
-	params.Namespace.BdevName = in.GetNamespace().GetBdev()
+	// TODO: using bdev as malloc until middlelayer reference is sorted out
```
Comment (glimchb) [inline server/frontend.go:188]: we have implantation today they relies on passing BDEV... 
so please leave the TODO comnent...
but let's use `CryptoKeyId.Value`  as a workaround to pass the desired BDEV... 
That's why I didn't want to delete the `bdev` from the front end...

File: server/frontend.go [hunk 3]
```diff
@@ -170,71 +225,103 @@ func (s *server) NVMeNamespaceDelete(ctx context.Context, in *pb.NVMeNamespaceDe
...
-	log.Printf("Received from client: %v", in.GetNamespace())
+	log.Printf("Received from client: %v", in.Namespace)
+	namespaces[in.Namespace.Id.Value] = *in.Namespace
 	return &pb.NVMeNamespaceUpdateResponse{}, nil
 }
 
 func (s *server) NVMeNamespaceList(ctx context.Context, in *pb.NVMeNamespaceListRequest) (*pb.NVMeNamespaceListResponse, error) {
 	log.Printf("NVMeNamespaceList: Received from client: %v", in)
+	
+	nqn := "all"
```
Comment (glimchb) [inline server/frontend.go:241]: either `all` or empty `SubsystemId` can indicate we want everything

**Review by glimchb — CHANGES_REQUESTED** [review]
please fix `BDEV` as a must, all the rest are small comments for your consideration

File: server/frontend.go (hunk 2 above)
Comment (jainvipin) [reply server/frontend.go:188]: ack; fixed.

File: server/frontend.go (hunk 3 above)
Comment (jainvipin) [reply server/frontend.go:241]: ack; fixed.

File: client/frontend.go (hunk 1 above)
Comment (jainvipin) [reply client/frontend.go:11]: I kept them somewhat morel legible for easy debugging, let's discuss...

**Review by glimchb — APPROVED** [review]

**Pull request merged**